packages/graphiql-react/src/editor/completion.ts

```diff
@@ -0,0 +1,107 @@
+import type { Editor, EditorChange } from 'codemirror';
```

Comment:
Copied from `packages/graphiql/src/utility/onHasCompletion.ts`

Reply:
Is there a reason why the files are treated as new and not index changes? Is it because there are lots of changed lines? Or perhaps because the files aren’t deleted yet? Sorry reviewing from phone. Index change is ideal to retain git history, but if not, let’s have comments at the top of the file to link to to permalinks to the legacy files at least.

Reply:
My goal was to move each of these `@graphiql/react` iterations would be a set of git mv’d files from `graphiql` workspace, but that’s just what I envisioned

Reply:
The files in `graphiql` aren't deleted yet!

I just moved the logic for the header editor in this PR to keep the LOCs changed in check 😅 So we need them in both places atm. Once we moved over the other editors (already working on it) we can remove the files from `graphiql`.